### PR TITLE
[ADD] Add account_move_reconcile_helper module

### PR DESCRIPTION
**Pull request by adrienpeiffer**
This module was written to offer facilities about reconciliation :
- Adding color for reconciliation on Journal items view
- Add the possibility to group by reconcilation on journal items view

### CODE REVIEW

**Comment by sbidoul** [comment]
@adrienpeiffer the method for coloring the reoncile_ref cells should be factored out in an OCA/web module named web_widget_auto_color with the purpose of giving a visual cue to cells having the same value on screen.

Then this PR can be updated by simply using this widget on the reconcile_ref field.

**Comment by adrienpeiffer** [comment]
I don't really understand why runbot fail but travis is green.

**Comment by sbidoul** [comment]
@adrienpeiffer can you rebase (there is a conflict in travis.yml)?

File: account_move_reconcile_helper/README.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+account_move_reconcile_helper
+=============================
+
+This module was written to offer facilities about reconciliation
```
Comment (pedrobaeza) [inline account_move_reconcile_helper/README.rst]: This text doesn't correspond to the module content

Comment (adrienpeiffer) [inline account_move_reconcile_helper/README.rst]: @pedrobaeza What do you mean exactly ?

**Comment by sbidoul** [comment]
:+1: (code review and functional test)

File: account_move_reconcile_helper/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with account_move_reconcile_helper.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': "Account Move Reconcile Helper",
+    'author': "ACSONE SA/NV",
```
Comment (lmignon) [inline account_move_reconcile_helper/__openerp__.py]: OCA is missing as author

File: account_move_reconcile_helper/README.rst (hunk 1 above)
Comment (pedrobaeza) [inline account_move_reconcile_helper/README.rst]: Sorry, I only see the model part. Anyways, please add that also include a field for the balance of the line (credit_debit_balance)

File: .travis.yml [hunk 3]
```diff
@@ -27,6 +27,8 @@ virtualenv:
   system_site_packages: true
 
 install:
+  - git clone https://github.com/OCA/reporting-engine ${HOME}/reporting-engine -b ${VERSION}
+  - git clone https://github.com/OCA/web ${HOME}/web -b ${VERSION}
```
Comment (lmignon) [inline .travis.yml]: You can put your dependencies in a file named `oca_dependencies.txt` (sample https://github.com/acsone/l10n-belgium/blob/8.0-transifex-sbi/oca_dependencies.txt)

File: .travis.yml [hunk 4]
```diff
@@ -27,6 +27,8 @@ virtualenv:
...
   - git clone https://github.com/oca/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
```
Comment (lmignon) [inline .travis.yml]: you can add `--depth 1` -> `git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools`

**Comment by lmignon** [comment]
:+1:  Once Travis is green (code review)

**Comment by adrienpeiffer** [comment]
@pedrobaeza @lmignon @sbidoul Thanks for the review !

**Comment by luc-demeyer** [comment]
I am moving the discussion started in PR 110 to this PR since 110 is merged and closed in the meanwhile.

I have a preference for a new module that combines the functional features of account_move_line_search_extension, account_move_reconcile_helper and account_move_line_balance.

A second best option is stephane's suggestion to adapt account_move_line_balance to run on top of both modules. But having three modules going to the same target customers and serving largely the same set of use cases doesn't really help the widespread use of these modules into the community.

My proposal:

I can make a new module 'account_move_line_search_and_reconcile_helper' (suggestions for another name are welcomed). This module combines all features of the three modules and adds the 'absolute balance' field to the 'journal items search all' search boxes (after the period box)

Remaining question is how to implement the single click to sort on absolute balance
- the easiest way to implement is to show both balance and absolute balance on the user interface,but I don't think end users will like this approach
- another option is to make absolute balance an invisible field and do some javascript (e.g. intercept the click on the 'balance' column or add a button to sort on 'absolute balance').

Any feedback is welcomed.

**Comment by pedrobaeza** [comment]
We can also hack the corresponding order method for getting correctly ordered the non absolute balance column without the need of the other one.

**Comment by pedrobaeza** [comment]
Are you still interested in this? CIs status is red.

**Comment by sbidoul** [comment]
Most of our 8.0 customers use it. I rebased to see what travis says.

**Comment by lmignon** [comment]
@sbidoul @pedrobaeza travis is :green_apple:  :tada:

**Comment by pedrobaeza** [comment]
Runbot is not, but it's a different history (incompatibility between account_asset and account_asset_management). Then I proceed to merge this one optimistically as it comes from a trusted source (my goal is to reduce the number of opened PRs as you can see due to my recent activity).

**Pull request merged**